src/components/SettingsModal.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useContext, useEffect } from "react";
+import reactDom from "react-dom";
+import classnames from "classnames";
+import { v4 as uuidv4 } from "uuid";
+import { ModalContext } from "./ClockFace.js"
+import "./SettingsModal.scss";
+
+const SettingsModal = () => {
+  const { isModalOpen, setIsModalOpen, sessions, setSessions } = useContext(ModalContext);
+  const [localSession, setLocalSession] = useState(sessions);
+  const [hasSettingsChanged, setHasSettingsChanged] = useState(false);
+
+  useEffect(() => {
+    if (isModalOpen) {
+      setLocalSession(sessions);
+      setHasSettingsChanged(false);
+    }
+  }, [isModalOpen, sessions])
+
+  if (!isModalOpen) {
+    return null;
+  }
+
+  const applyAndClose = () => {
+    setSessions(localSession)
+    setIsModalOpen(false);
+  }
+
+  const handleInputChange = (e) => {
+    if (e.target.valueAsNumber < 1 || e.target.valueAsNumber > 60 || !e.target.valueAsNumber) {
+      return false;
+    }
+    else {
+      setHasSettingsChanged(true);
+      if (e.target.name === "longSession") {
+        const newSettings = [...localSession];
+        newSettings[0] = e.target.valueAsNumber * 60;
+        newSettings[2] = e.target.valueAsNumber * 60;
+        setLocalSession(newSettings);
+      }
+      else if (e.target.name === "shortSession") {
+        const newSettings = [...localSession];
+        newSettings[1] = e.target.valueAsNumber * 60;
+        newSettings[3] = e.target.valueAsNumber * 60;
+        setLocalSession(newSettings);
```

Comment:
This doesn't look pretty, but this works and I have looked and this is how to update an array state at specific indices.